### DISCUSSION TITLE
How to represent QSA adapter?

### DISCUSSION THREAD

**Discussion by sashabovs** [netbox-community/netbox#18052] [Q&A]
Hi everyone,
I'm looking for possible ways to represent the QSA adapter (similar to [this](https://www.sfpcables.com/40g-to-10g-qsfp-adapter-qsa-40-version-a-default)) in Netbox.
In my company we have a QSFP interface with a QSA adapter in it. Then we put SFP+ transceiver and then connect other device to the transceiver via cable.

I have an idea to make them as modules. 
Then we have QSFP interface as Module bay. In it we put QSA adapter (as module) also with module bay. And then we put SFP+ transceiver (as module) with a Netbox interface in it.
But the problem with modules is, that we will not be able to use devices from Device Library (since the QSFP interfaces will need to be manually converted into module bays). And if I want to move the module, I will have to delete old one and create it again entering all data one more time.

I can make QSA adapter a device. But in this case the interface of this device will not be seen directly from the parent device. And it doesn't seam to be natural to present an adapter as device. I would like not to see it in the device list. 
Another problem is, that we can have QSFP interface on module. In this case we cannot connect QSA device using device bay. We will need to create a "virtual" cable between QSFP and QSA adapter. 

What is the best way to represent this structure in Netbox?

**Comment by candlerb** [comment]
Even regular SFPs aren't explicitly modelled in Netbox.

What I suggest you do is:
- terminate cables directly in the device's interface, as if there were no SFP
- add the SFP, and also the QSA adapter in this case, as Inventory Items, which can be associated with the Interface

See the section "Component Assignment" when you create an Inventory Item
